### ISSUE TITLE
Add basics for implementing app in README/Wiki

### ISSUE THREAD

**Issue by blasterpal** [blakechambers/matterhorn#42]
We should extract the basic steps to setup an app using Gem and using the basic matchers to test implementing app. 

Example: 

### Add to your Gemfile
`gem matterhorn `

### Generate a basic app based on Rails-api
```
rails-api new my_json_api_app
```

### Create Matterhorn Controller
```
class PostsController < Matterhorn::Base
   include Matterhorn::Resources
```

And so forth

**Comment by blasterpal** [comment]
Started some of this in https://github.com/blakechambers/matterhorn/pull/44

**Comment by blakechambers** [comment]
While this is somewhat helpful, I think it's really just a start of what a guide could/should be.

I think the right time to write guides like this would be _after_ the core functionality is completed (aka 0.1.0) and after the [DSL cleanup 0.2.0](https://github.com/blakechambers/matterhorn/blob/master/ROADMAP.md#020---dsl-cleanup) version of the gem.  For now, I'd find more value in an example application that people can clone and start playing around with.

I'll add a note to the 0.2.0 about that, otherwise considering this closed until after 0.2.0 is finished.
